### PR TITLE
feat: support search with lang cyrillic

### PR DESCRIPTION
**Pull request by ssoda01**
## Summary
support search with language:cyrillic

## Related Issue
 #239

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: .changeset/loud-rules-hang.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@rspress/theme-default": major
```
Comment (Timeless0911) [inline .changeset/loud-rules-hang.md:2]: maybe a minor change is better

File: packages/theme-default/src/components/Search/logic/providers/LocalProvider.ts [hunk 2]
```diff
@@ -76,8 +79,23 @@ export class LocalProvider implements Provider {
         return cjkWords;
       },
     });
+    this.#cyrilicIndex = FlexSearch.create({
+      ...createOptions,
+      tokenize(str: string) {
```
Comment (Timeless0911) [inline packages/theme-default/src/components/Search/logic/providers/LocalProvider.ts:84]: The tokenize method can be abstracted for better maintenance in the future~

File: .changeset/loud-rules-hang.md (hunk 1 above)
Comment (ssoda01) [reply .changeset/loud-rules-hang.md]: Thank you for pointing out the error in this and helping me to fix it, it has been fixed

File: packages/theme-default/src/components/Search/logic/providers/LocalProvider.ts (hunk 2 above)
Comment (ssoda01) [reply packages/theme-default/src/components/Search/logic/providers/LocalProvider.ts]: Thank you for your help in pointing out the shortcomings and teaching me to revise it. It has been fixed.

**Comment by Timeless0911** [comment]
Thank you for your awesome contribution~

**Comment by sanyuan0704** [comment]
LGTM

**Review by sanyuan0704 — APPROVED** [review]

**Pull request merged**